hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBScanner.java

```diff
@@ -136,209 +154,168 @@ public static byte[] getValueObject(
     DBDefinitionFactory.setDnDBSchemaVersion(dnDBSchemaVersion);
-    this.constructColumnFamilyMap(DBDefinitionFactory.
-            getDefinition(Paths.get(dbPath), new OzoneConfiguration()));
-    if (this.columnFamilyMap != null) {
-      if (!this.columnFamilyMap.containsKey(tableName)) {
-        System.out.print("Table with name:" + tableName + " does not exist");
-      } else {
-        DBColumnFamilyDefinition columnFamilyDefinition =
-                this.columnFamilyMap.get(tableName);
-        ColumnFamilyHandle columnFamilyHandle = getColumnFamilyHandle(
-                columnFamilyDefinition.getTableName()
-                        .getBytes(StandardCharsets.UTF_8),
-                columnFamilyHandleList);
-        if (columnFamilyHandle == null) {
-          throw new IllegalArgumentException("columnFamilyHandle is null");
-        }
-        if (showCount) {
-          long keyCount = rocksDB.get().getLongProperty(columnFamilyHandle,
-              RocksDatabase.ESTIMATE_NUM_KEYS);
-          System.out.println(keyCount);
-          return;
-        }
-        ManagedRocksIterator iterator;
-        if (containerId > 0 && dnDBSchemaVersion != null &&
-            dnDBSchemaVersion.equals("V3")) {
-          ManagedReadOptions readOptions = new ManagedReadOptions();
-          readOptions.setIterateUpperBound(new ManagedSlice(
-              FixedLengthStringUtils.string2Bytes(
-                  DatanodeSchemaThreeDBDefinition.getContainerKeyPrefix(
-                  containerId + 1))));
-          iterator = new ManagedRocksIterator(
-              rocksDB.get().newIterator(columnFamilyHandle, readOptions));
-          iterator.get().seek(FixedLengthStringUtils.string2Bytes(
+    DBDefinition dbDefinition = DBDefinitionFactory.getDefinition(
+        Paths.get(dbPath), new OzoneConfiguration());
+    if (dbDefinition == null) {
+      err().println("Error: Incorrect DB Path");
+      return;
+    }
+
+    Map<String, DBColumnFamilyDefinition> columnFamilyMap = new HashMap<>();
+    for (DBColumnFamilyDefinition cfDef : dbDefinition.getColumnFamilies()) {
+      LOG.info("Found table: {}", cfDef.getTableName());
+      columnFamilyMap.put(cfDef.getTableName(), cfDef);
+    }
+    if (!columnFamilyMap.containsKey(tableName)) {
+      err().print("Error: Table with name '" + tableName + "' not found");
+      return;
+    }
+
+    DBColumnFamilyDefinition columnFamilyDefinition =
+        columnFamilyMap.get(tableName);
+    ColumnFamilyHandle columnFamilyHandle = getColumnFamilyHandle(
+        columnFamilyDefinition.getTableName().getBytes(UTF_8),
+        columnFamilyHandleList);
+    if (columnFamilyHandle == null) {
+      throw new IllegalStateException("columnFamilyHandle is null");
+    }
+
+    if (showCount) {
+      long keyCount = rocksDB.get().getLongProperty(columnFamilyHandle,
+          RocksDatabase.ESTIMATE_NUM_KEYS);
+      out().println(keyCount);
+      return;
+    }
+    ManagedRocksIterator iterator;
```

Comment:
Not sure, how big it is a concern for CLI, but as general practice, `ManagedRocksIterator` should be closed after the use.